R/Sharis/Ameriflux_functions.R

```diff
@@ -0,0 +1,92 @@
+generate_attributes_table <- function(csv_file_path,
+                                      attributes_file_path) {
+  # Check that files exist 
+  stopifnot(file.exists(csv_file_path))
+  stopifnot(file.exists(attributes_file_path))
+  
+  # Read in files 
+  data <- read.csv(csv_file_path, stringsAsFactors = FALSE, skip = 2)
+  n <- dim(data)[2]
+  attributes <- try(read.csv(attributes_file_path, stringsAsFactors = FALSE))
+
+  
+  # Initialize data frame
+  table <- data.frame(attributeName = rep("NA", n),
+                      attributeDefinition = rep("NA", n),
+                      measurementScale = rep("NA", n),
+                      domain = rep("NA", n),
+                      formatString = rep("NA", n),
+                      definition = rep("NA", n),
+                      unit = rep("NA", n),
+                      numberType = rep("NA", n),
+                      missingValueCode = rep("NA", n),
+                      missingValueCodeExplanation = rep("NA", n),
+                      stringsAsFactors = F)
+  
+  qualifiers<- c("_PI", "_QC", "_F", "_IU", "_H_V_R", "_H_V_A", "_1", "_2", "_3", "_4", "_5", "_6", "_7", "_8", "_9", "_SD", "_N")
+  num_qualifiers<- c("_1", "_2", "_3", "_4", "_5", "_6", "_7", "_8", "_9")
+  
+  for (i in seq_len(n)) {
+    # add attribute name
+    table$attributeName[i] = colnames(data)[i]
+    
+      ## check if the name has a qualifier at the end
+      if (any(endsWith(colnames(data)[i], suffix = qualifiers))) {
+        # identify the qualifier
+        current_qual <- which(endsWith(colnames(data)[i], suffix = qualifiers))
+        qualifier<- qualifiers[current_qual]
+        len<- nchar(qualifier)
+        main_label<- substr(colnames(data)[i], 1, nchar(colnames(data)[i])-len)
+        
+        # get definition for main label
+        main_def <- attributes$uniqueAttributeDefinition[attributes$uniqueAttributeLabel == main_label]
```

Comment:
at the beginning of your function you could specify `colnames(attributes) <- ("category", "label", "definition", "unit", "SI_unit")` and get rid "uniqueAttribute" everywhere in your code.  

The benefit is making your code more readable - also not your fault, the column names from that csv aren't great.